incubator/metrics-http/src/main/java/io/aklivity/zilla/runtime/metrics/http/internal/HttpDurationMetricContext.java

```diff
@@ -100,28 +102,41 @@ public void accept(
             switch (msgTypeId)
             {
             case BeginFW.TYPE_ID:
-                if (direction == 1L) //received
-                {
-                    timestamps.put(exchangeId, System.currentTimeMillis());
-                }
+                handleBegin(exchangeId, direction, frame.timestamp());
                 break;
             case ResetFW.TYPE_ID:
             case AbortFW.TYPE_ID:
             case EndFW.TYPE_ID:
-                final long mask = 1L << direction;
-                final long status = exchanges.get(exchangeId) | mask; // mark current direction as closed
-                if (status == EXCHANGE_CLOSED) // both received and sent streams are closed
+                handleEnd(msgTypeId, exchangeId, direction, frame.timestamp());
```

Comment:
Please keep the implementation inlined. 

Please also record only duration for requests that `END` successfully in both directions, per file comment.

Reply:
OK